### ISSUE TITLE
Sentence Embeddings using Siamese BERT-Networks

### ISSUE THREAD

**Issue by dltmddbs100** [dltmddbs100/Paper-log#21]
## Main Idea
**Derive semantically meaningful sentence embedding**
Paper: <https://arxiv.org/abs/1908.10084>

## Abstract
BERT (Devlin et al., 2018) and RoBERTa (Liu et al., 2019) has set a new state-of-the-art performance on sentence-pair regression tasks like semantic textual similarity (STS). However, it requires that both sentences are fed into the network, which causes a massive computational overhead: Finding the most similar pair in a collection of 10,000 sentences requires about 50 million inference computations (~65 hours) with BERT. The construction of BERT makes it unsuitable for semantic similarity search as well as for unsupervised tasks like clustering.
In this publication, we present Sentence-BERT (SBERT), a modification of the pretrained BERT network that use siamese and triplet network structures to derive semantically meaningful sentence embeddings that can be compared using cosine-similarity. This reduces the effort for finding the most similar pair from 65 hours with BERT / RoBERTa to about 5 seconds with SBERT, while maintaining the accuracy from BERT. We evaluate SBERT and SRoBERTa on common STS tasks and transfer learning tasks, where it outperforms other state-of-the-art sentence embeddings methods.

## Contribution
+ BERT가 갖는 sentence embedding의 시간적인 비효율성, 성능 저하를 극복
+ Siamese network의 구조로 pre-trained BERT 또는 RoBERTa에 고정된 vector를 출력하는 pooling operation을 추가한 형태
+ 두 개의 input을 동시에 받아, 동일한 parameter를 갖는 BERT를 거쳐 pooling layer를 통해 각각의 fixed size sentence embedding에 해당하는 u, v를 반환
+ classification, regression, triplet의 다양한 objective function 사용
+ MEAN pooling strategy,  (u, v, |u-v|) concatenation이 가장 효과적 => 특히 |u-v| term이 중요
+ SentEval task에서 sota 달성 => semantic sentence embedding 형성
+ 연산적인 측면에서 효율적, clustering과 같은 task도 쉽게 수행가능

## Reviews
<https://deepkerry.tistory.com/34>
